src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,32 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+interface Props {
+  todo: Todo;
+  onModalClose: () => void;
+}
+
+export const TodoModal: React.FC<Props> = ({ todo, onModalClose }) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  const [isUserLoaded, setIsUserLoaded] = useState(false);
+
+  useEffect(() => {
+    getUser(todo.userId).then((fetchedUser: User) => {
+      setUser(fetchedUser);
+
+      setIsUserLoaded(true);
+    });
+  }, []);
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
+      {!isUserLoaded ? (
```

Comment:
I suggest to do it like this:

`      {user ? (
                <a href={user.email}>{user.name}</a>
              ) : (
                <p>The user is not found</p>
              )}`
              
  Because if we add a check here 
  
  `{!isUserLoaded && !user ? (`
  
   there is a possibility of the infinite loader which will be impossible to close for user